### PR TITLE
Remove dead code

### PR DESCRIPTION
**Pull request by skeleton**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

During my POC about permissions, I found dead code. To simplify our refacto on permissions, I remove it right now.

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Review by grena — APPROVED** [review]
Not even used in EE?
If tests are ok well.. GTM ^^

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/CurrencyRepository.php [hunk 1]
```diff
@@ -45,28 +45,28 @@ public function getActivatedCurrencyCodes()
     /**
      * {@inheritdoc}
      */
-    public function getActivatedCurrenciesQB()
```
Comment (SamirBoulil) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/CurrencyRepository.php]: \o/

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/GroupRepository.php [hunk 2]
```diff
@@ -231,57 +80,55 @@ public function getOptions($dataLocale, $collectionId = null, $search = '', arra
...
-    protected function getGroupsByType(GroupTypeInterface $type)
-    {
-        return $this
-            ->getGroupsByTypeQB($type)
-            ->getQuery()
-            ->getResult();
+        $isCheckecExpr =
+            'CASE WHEN (g.id IN (:associatedIds) OR g.id IN (:data_in)) AND g.id NOT IN (:data_not_in) ' .
+            'THEN true ELSE false END';
+
```
Comment (SamirBoulil) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/GroupRepository.php:116]: extra line (don't *need* to fix, just saying ^^)

File: src/Pim/Bundle/EnrichBundle/Controller/ProductController.php [hunk 3]
```diff
@@ -127,7 +127,7 @@ public function __construct(
      *
      * @return array
      */
-    public function indexAction()
+    public function indexAction(Request $request)
```
Comment (SamirBoulil) [inline src/Pim/Bundle/EnrichBundle/Controller/ProductController.php:130]: Are we sure ?

**Review by SamirBoulil — APPROVED** [review]
SOOO Much ❤️  cleanup 💃  !! Nice PR

**Review by ahocquard — APPROVED** [review]
🔥 🔥

**Review by a2xchip — APPROVED** [review]
:) Looks amazing! Could not imagine there are so many useless methods

**Comment by skeleton** [comment]
@grena CI is still running. But methods do not exist anymore in the code. Maybe some of these methods are called magically or with a concatenation of variable, but CI will give results in few minutes :)

File: src/Pim/Bundle/EnrichBundle/Controller/ProductController.php (hunk 3 above)
Comment (skeleton) [reply src/Pim/Bundle/EnrichBundle/Controller/ProductController.php:130]: Yes, this method is overriden on EE. As signature is not the same, there is a warning on EE

**Review by juliensnz — APPROVED** [review]

**Pull request merged**